boot/boottest/device.go

```diff
@@ -60,9 +59,16 @@ func snapAndMode(str string) (snap, mode string) {
 	return parts[0], parts[1]
 }
 
-func (d *mockDevice) Kernel() string        { return d.bootSnap }
-func (d *mockDevice) Base() string          { return d.bootSnap }
-func (d *mockDevice) Classic() bool         { return d.bootSnap == "" }
-func (d *mockDevice) RunMode() bool         { return d.mode == "run" }
-func (d *mockDevice) HasModeenv() bool      { return d.uc20 }
-func (d *mockDevice) Model() *asserts.Model { return d.model }
+func (d *mockDevice) Kernel() string   { return d.bootSnap }
+func (d *mockDevice) Base() string     { return d.bootSnap }
+func (d *mockDevice) Classic() bool    { return d.bootSnap == "" }
+func (d *mockDevice) RunMode() bool    { return d.mode == "run" }
+func (d *mockDevice) HasModeenv() bool { return d.uc20 }
+func (d *mockDevice) Model() *asserts.Model {
+	if d.model == nil {
+		panic("tried to use model without setting one first")
+	}
+	return d.model
+}
+
+func (d *mockDevice) SetModel(model *asserts.Model) { d.model = model }
```

Comment:
we discussed this, but in hindsight it's confusing because  then Kernel and Base return get out of sync. I proposed #9339 instead